src/mlpack/tests/gan_test.cpp

```diff
@@ -166,6 +168,10 @@ BOOST_AUTO_TEST_CASE(GANMNISTTest)
 
   trainData = trainData.cols(0, datasetMaxCols - 1);
 
+  // Normalize the input data.
+  trainData = trainData - 127.5;
```

Comment:
Is `127.5` equivalent to `arma::mean(trainData, 1)`? Might be more clear to use `mean` instead of a const value.

Reply:
No, actually as we are using TanHLayer in the generator network our fake images will have pixels in the range of `[-1, 1]` so we need to scale our input images to `[-1, 1]` as well. 
It's mention below as first point in `readme.md` as first point.
https://github.com/soumith/ganhacks